### PR TITLE
MULE-10466: Update javax transport version to 1.2

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: distributions/standalone/assembly-whitelist.txt [hunk 1]
```diff
@@ -296,6 +295,7 @@
 /mule-standalone-${productVersion}/lib/opt/javassist-3.7.ga.jar
 /mule-standalone-${productVersion}/lib/opt/javax.annotation-api-1.2.jar
 /mule-standalone-${productVersion}/lib/opt/javax.inject-1.jar
+/mule-standalone-${productVersion}/lib/opt/javax.transaction-api-1.2-b03.jar
```
Comment (pablokraan) [inline distributions/standalone/assembly-whitelist.txt]: This version is a beta, use 1.2

Comment (mbuchwald) [inline distributions/standalone/assembly-whitelist.txt]: Fixed

**Comment by aiannucci** [comment]
There should be an equivalent EE PR for this change.

**Comment by mbuchwald** [comment]
Done

File: pom.xml [hunk 2]
```diff
@@ -575,10 +575,6 @@
                     </exclusion>
                     <exclusion>
                         <groupId>org.apache.geronimo.specs</groupId>
-                        <artifactId>geronimo-jta_1.0.1B_spec</artifactId>
```
Comment (aiannucci) [inline pom.xml]: If you remove this exclusion it would cause the library to be potentially pulled again by the activemq dependency

Comment (mbuchwald) [inline pom.xml]: Fixed

**Pull request merged**